### PR TITLE
new rule: no anyOf & oneOf in OpenAPI v2

### LINKED ISSUE #393: new rules: error for anyOf & oneOf in OpenAPI v2
**User story.**

As an average user who is not completely familiar with the intricacies of JSON Schema <≠> OpenAPI and the differences between OpenAPI v2 and v3, I often mix up JSON Schema, OAS2 and OAS3 keywords like anyOf and oneOf, so please let me know when I put the wrong keyword in the wrong place.

**Is your feature request related to a problem?**

API description documents should conform to the description format in question, as putting invalid keywords in those documents can lead to inconsistent behavior across various tools that might mistakenly support those keywords. Other tooling which is actually compliant will then look like its failing.

**Describe the solution you'd like**

The rule should error if an `anyOf`, `oneOf` or `not` appears in OpenAPI v2, but `anyOf` is in fact allowed.

### PR DESCRIPTION
**Pull request by philsturgeon**
Fixes #393.

**Checklist**

- [x] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Screenshots**

Screen Shot 2019-07-26 at 15 31 33

**Additional context**

### CODE REVIEW

File: src/rulesets/oas2/index.json [hunk 1]
```diff
@@ -103,8 +103,36 @@
...
+        "field": "anyOf",
+        "function": "undefined"
+      },
+      "tags": [
+        "schema"
+      ]
+    },
+    "oas2-oneOf": {
+      "description": "OpenAPI v3 keyword `oneOf` detected in OpenAPI v2 document.",
+      "message": "anyOf is not available in OpenAPI v2, it was added in OpenAPI v3",
```
Comment (P0lip) [inline src/rulesets/oas2/index.json]:
```suggestion
      "message": "oneOf is not available in OpenAPI v2, it was added in OpenAPI v3",
```

File: src/cli/commands/__tests__/lint.test.ts [hunk 2]
```diff
@@ -337,47 +337,49 @@ describe('lint', () => {
...
-          }),
-          expect.objectContaining({
-            code: 'oas2-schema',
-            message: '/info Property foo is not expected to be here',
-            path: ['info'],
-            range: {
-              end: {
-                character: 5,
-                line: 9,
+        expect(JSON.parse(ctx.stdout)).toEqual(
```
Comment (P0lip) [inline src/cli/commands/__tests__/lint.test.ts]: Let's rever this change.

File: src/functions/__tests__/falsy.test.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+describe('falsy', () => {
+  test('returns undefined if target value is falsy', () => {
+    expect(runFalsy(false)).toBeUndefined();
+  });
+
+  test('returns undefined if target value is null', () => {
+    expect(runFalsy(null)).toBeUndefined();
+  });
+
+  test('returns error message if target value is falsy', () => {
```
Comment (P0lip) [inline src/functions/__tests__/falsy.test.ts]:
```suggestion
  test('returns error message if target value is not falsy', () => {
```

File: src/cli/commands/__tests__/lint.test.ts (hunk 2 above)
Comment (philsturgeon) [reply src/cli/commands/__tests__/lint.test.ts]: Why? Reasons for things should be included on review comments. We gotta know why we're doing stuff.

File: src/functions/__tests__/falsy.test.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    expect(runFalsy(true)).toEqual([
+      {
+        message: 'property is not falsy',
+      },
+    ]);
+  });
+
+  test('returns undefined if target path is set', () => {
```
Comment (P0lip) [inline src/functions/__tests__/falsy.test.ts]: Let's add another test case with a path.

File: src/cli/commands/__tests__/lint.test.ts (hunk 2 above)
Comment (P0lip) [reply src/cli/commands/__tests__/lint.test.ts]: You haven't touched any of existing fixtures, therefore this change should not be needed.
We actually did change the assertion here, as toEqual array might produce different results than  toEqual array containing (which IIRC matches just a subset of an array and items can be placed in any order)

**Pull request merged**